packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -214,6 +216,61 @@ export default class MongoBinaryDownload {
     return extractDir;
   }
 
+  async extractTarGz(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    const extract = tar.extract();
+    extract.on('entry', (header, stream, next) => {
+      if (filter(header.name))
+        stream.pipe(
+          fs.createWriteStream(path.resolve(extractDir, path.basename(header.name)), {
+            mode: 0o775,
+          })
+        );
+
+      stream.on('end', () => next());
+      stream.resume();
+    });
+
+    return new Promise((resolve, reject) => {
+      fs.createReadStream(mongoDBArchive)
+        .pipe(createUnzip())
+        .pipe(extract)
+        .on('end', () => resolve())
+        .on('error', (e) => reject(e));
+    });
+  }
+
+  async extractZip(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      yauzl.open(mongoDBArchive, { lazyEntries: true }, (e, zipfile) => {
+        if (e || !zipfile) reject(e);
+        zipfile?.readEntry();
+
+        zipfile?.on('end', () => resolve());
+
+        zipfile?.on('entry', (entry) => {
+          if (!filter(entry.fileName)) return zipfile.readEntry();
+          zipfile.openReadStream(entry, (e, r) => {
+            if (e || !r) reject(e);
+            r?.on('end', () => zipfile.readEntry());
```

Comment:
because `r` is existing please remove `?` (and if typescript says it dosnt, surround `reject` with "{}" and add an `return`